lib/src/position.dart

```diff
@@ -1720,6 +1721,140 @@ class ThreeCheck extends Position<ThreeCheck> {
   }
 }
 
+/// A variant where the goal is to put your king on the eigth rank
+@immutable
+class RacingKings extends Position<RacingKings> {
+  const RacingKings({
+    required super.board,
+    super.pockets,
+    required super.turn,
+    required super.castles,
+    super.epSquare,
+    required super.halfmoves,
+    required super.fullmoves,
+  });
+
+  const RacingKings._initial()
+      : super(
+            board: Board.racingKings,
+            pockets: null,
+            turn: Side.white,
+            castles: Castles.standard,
+            epSquare: null,
+            halfmoves: 0,
+            fullmoves: 1);
+
+  static const initial = RacingKings._initial();
+  static const goal = SquareSet.fromRank(7);
+
+  bool get blackCanReachGoal {
+    final blackKing = board.kingOf(Side.black);
+    return blackKing != null &&
+        !kingAttacks(blackKing).intersect(goal).squares.where((square) {
+          // Check whether this king move is legal
+          final context = _Context(
+            isVariantEnd: false,
+            mustCapture: false,
+            king: blackKing,
+            blockers: _sliderBlockers(blackKing),
+            checkers: checkers,
+          );
+          final legalMoves = _legalMovesOf(blackKing, context: context);
+          return legalMoves.has(square);
+        }).isEmpty;
+  }
+
+  bool get blackInGoal =>
+      !board.black.intersect(goal).intersect(board.kings).isEmpty;
+  bool get whiteInGoal =>
+      !board.white.intersect(goal).intersect(board.kings).isEmpty;
+
+  @override
+  bool get isVariantEnd {
+    if (!whiteInGoal && !blackInGoal) {
+      return false;
+    }
+    if (blackInGoal || !blackCanReachGoal) {
+      return true;
+    }
+    return false;
+  }
+
+  @override
+  Outcome? get variantOutcome {
+    if (!isVariantEnd) return null;
+    if (whiteInGoal && blackInGoal) return Outcome.draw;
+    final whiteKing = board.kingOf(Side.white);
+    // If white is in the goal, check
+    // whether black can reach the goal
+    // And vice-versa
+    if (whiteInGoal && !blackCanReachGoal) return Outcome.whiteWins;
+    if (blackInGoal) return Outcome.blackWins;
+
+    return Outcome.draw;
+  }
+
+  /// Set up a playable [RacingKings] position.
+  ///
+  /// Throws a [PositionError] if the [Setup] does not meet basic validity
+  /// requirements.
+  /// Optionnaly pass a `ignoreImpossibleCheck` boolean if you want to skip that
+  /// requirement.
+  factory RacingKings.fromSetup(Setup setup, {bool? ignoreImpossibleCheck}) {
+    final pos = RacingKings(
+      board: setup.board,
+      turn: setup.turn,
+      castles: Castles.fromSetup(setup),
```

Comment:
RacingKings doesn't support either castling or enpassant, so, I think we should either ignore them, or return errors if the setup has them.